### ISSUE TITLE
Floating-point exception with SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/E3SM#5463]
Error with this test `SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu`

```

3097: Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.
3097: 
3097: Backtrace for this error:
3097: #0  0x15554ed3a3ff in ???
3097: #1  0x4760382 in compute_barycenter_coordinates
3097:   at /lcrc/group/e3sm/ac.ndkeen/scratch/chrys/SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu.20230217_143908_w0vt3t/bld/cmake-bld/core_seaice/shared/mpas_seaice_advection_incremental_remap.f90:4772
3097: #2  0x477de30 in construct_linear_tracer_fields
3097:   at /lcrc/group/e3sm/ac.ndkeen/scratch/chrys/SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu.20230217_143908_w0vt3t/bld/cmake-bld/core_seaice/shared/mpas_seaice_advection_incremental_remap.f90:3881
3097: #3  0x4795459 in incremental_remap_block
3097:   at /lcrc/group/e3sm/ac.ndkeen/scratch/chrys/SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu.20230217_143908_w0vt3t/bld/cmake-bld/core_seaice/shared/mpas_seaice_advection_incremental_remap.f90:3067
3097: #4  0x4797110 in __seaice_advection_incremental_remap_MOD_seaice_run_advection_incremental_remap
3097:   at /lcrc/group/e3sm/ac.ndkeen/scratch/chrys/SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu.20230217_143908_w0vt3t/bld/cmake-bld/core_seaice/shared/mpas_seaice_advection_incremental_remap.f90:2575
3097: #5  0x3cd7244 in __seaice_advection_MOD_seaice_run_advection
3097:   at /lcrc/group/e3sm/ac.ndkeen/scratch/chrys/SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu.20230217_143908_w0vt3t/bld/cmake-bld/core_seaice/shared/mpas_seaice_advection.f90:138
3097: #6  0x3cd6fff in __seaice_time_integration_MOD_seaice_timestep
3097:   at /lcrc/group/e3sm/ac.ndkeen/scratch/chrys/SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu.20230217_143908_w0vt3t/bld/cmake-bld/core_seaice/shared/mpas_seaice_time_integration.f90:159
3097: #7  0x3b56f88 in __ice_comp_mct_MOD_ice_run_mct
3097:   at /lcrc/group/e3sm/ac.ndkeen/scratch/chrys/SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu.20230217_143908_w0vt3t/mpas-seaice/driver/ice_comp_mct.f90:1136
3097: #8  0x44e828 in __component_mod_MOD_component_run
3097:   at /lcrc/group/e3sm/ac.ndkeen/wacmy/m31-feb17/driver-mct/main/component_mod.F90:728
3097: #9  0x431ee7 in __cime_comp_mod_MOD_cime_run
3097:   at /lcrc/group/e3sm/ac.ndkeen/wacmy/m31-feb17/driver-mct/main/cime_comp_mod.F90:2897
3097: #10  0x44bd6d in cime_driver
3097:   at /lcrc/group/e3sm/ac.ndkeen/wacmy/m31-feb17/driver-mct/main/cime_driver.F90:153
3097: #11  0x44bdd4 in main
3097:   at /lcrc/group/e3sm/ac.ndkeen/wacmy/m31-feb17/driver-mct/main/cime_driver.F90:23


/lcrc/group/e3sm/ac.ndkeen/scratch/chrys/SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu.20230217_143908_w0vt3t

```

**Comment by jonbob** [comment]
@ndkeen, @rljacob - we have a fix for this, [PR #5252](https://github.com/E3SM-Project/E3SM/pull/5252). As soon as we get some critical fixes in, I'll make a longer run with the improved code to make sure any non-BFB behavior has minimal impact

**Comment by jonbob** [comment]
@ndkeen - I did run this test with a merge of the fix in [PR #5252](https://github.com/E3SM-Project/E3SM/pull/5252) and it passed:
````
PASS SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu CREATE_NEWCASE
PASS SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu XML
PASS SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu SETUP
PASS SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu SHAREDLIB_BUILD time=101
PASS SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu MODEL_BUILD time=371
PASS SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu SUBMIT
PASS SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu RUN time=6963
PASS SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu MEMLEAK
PASS SMS_D_P4096.ne120pg2_r0125_oRRS18to6v3.WCYCL1950.chrysalis_gnu SHORT_TERM_ARCHIVER
````
However, I did some other testing and the PR is non-BFB, so it will take a bit of work to get it merged
